src/main.ts

```diff
@@ -39,20 +40,15 @@ export function createEmitter<T extends Events>({ broadcastChannel: useBroadcast
   }
 
   function on(globalEventHandler: GlobalEventHandler): () => void
-  function on<E extends Event>(event: E, handler: Handler<T[E]>): () => void
-  function on<E extends Event>(globalHandlerOrEvent: E | GlobalEventHandler, handler?: Handler<T[E]>): () => void {
+  function on<E extends Event>(event: E, handler: Handler<EventPayload<E>>): () => void
+  function on<E extends Event>(globalHandlerOrEvent: E | GlobalEventHandler, maybeHandler?: Handler<EventPayload<E>>): () => void {
     if (isGlobalEventHandler(globalHandlerOrEvent)) {
       globalHandlers.add(globalHandlerOrEvent)
       
       return () => off(globalHandlerOrEvent)
     }
 
-    const event = globalHandlerOrEvent
-
-    if (!handler) {
-      throw new Error(`Handler must be given for ${String(event)} event`)
-    }
-
+    const { event, handler } = verifyEventAndHandler(globalHandlerOrEvent, maybeHandler)
```

Comment:
I think something like this is more straight forward even if its a little less DRY. That way these methods (which are pretty simple) contain all their own business logic and don't require reading utilities to understand what they do. 
```
const event = globalHandlerOrEvent

if(!isHandler(handler) {
  throw new MissingEventHandlerError(event)
}

on(event, handler)
```